plenum/server/node.py

```diff
@@ -2082,9 +2075,14 @@ def primary_selected(self, instance_id):
         if instance_id == 0:
             # TODO: 0 should be replaced with configurable constant
             self.monitor.hasMasterPrimary = self.primaryReplicaNo == 0
-
-        if self.view_change_in_progress and self.all_instances_have_primary:
-            self.on_view_change_complete(self.viewNo)
+        if self.all_instances_have_primary:
+            # Doing this for CurrentState message
+            # This problem is faced because viewNo is set on view change start
+            view_no = self.viewNo
```

Comment:
Node's value of `viewNo` is different from elector for a very short period of time (in startViewChange), and the only messages for elector are put on the queue and not sent yet.